source/d3d11/depth_counter_tracker.hpp

```diff
@@ -0,0 +1,33 @@
+#pragma once
+
+#include <d3d11.h>
+#include <unordered_map>
+
+class depth_counter_tracker
+{
+public:
+	depth_counter_tracker();
+	~depth_counter_tracker();
+
+	void reset();
+	void track_depthstencil(ID3D11DepthStencilView* to_track);
+	void log_drawcalls(ID3D11DepthStencilView* depthstencil, UINT drawcalls, UINT vertices);
+	ID3D11DepthStencilView* depth_counter_tracker::get_best_depth_stencil(UINT width, UINT height);
+	void merge(depth_counter_tracker& source);
+
+	UINT drawcalls() { return _drawcalls; }
+	void update_drawcalls(UINT amount) { _drawcalls += amount; }
+	UINT vertices() { return _vertices; }
+	void update_vertices(UINT amount) { _vertices += amount; }
+
+private:
+	struct depthstencil_counter_info
+	{
+		UINT drawcall_count = 0;
+		UINT vertices_count = 0;
+	};
+
+	std::unordered_map<ID3D11DepthStencilView*, depthstencil_counter_info> _counters_per_used_depthstencil;	
```

Comment:
I'd rather use a `com_ptr<ID3D11DeviceStencilView>` instead of a naked pointer to avoid manual reference counting (which is prone to errors). We can overload `std::hash<com_ptr<T>>` for this to work.

Reply:
As the pointer is here used just for lookup, it might be unnecessary to use com_ptr as well and simply don't do ref counting here? I think a refcount check then is needed in `draw_call_tracker::get_best_depth_stencil` to check if the buffer is still alive and we're not using a dangling pointer but that would be all? (also I don't know how to overload the hashfunction for com_ptr, but that's one google query away of course ;))

Reply:
Testing shows refcounters here are absolutely needed, memory leak is otherwise unavoidable when internal rendering resolution in e.g. AC:Origins is increased/decreased. Will look into hashfunction overrides for com_ptr

Reply:
Hash struct implemented. Hopefully I've done it properly (As I relied on a stackoverflow example for this) Tests show it makes sure everything is cleaned up properly, no ref counting needed :)